Content.Client/Clothing/NightVisionSystem.cs

```diff
@@ -0,0 +1,72 @@
+using Content.Shared.Clothing;
+using Content.Shared.GameTicking;
+using Robust.Client.Player;
+using Robust.Client.Graphics;
+using Content.Client.Inventory;
+using Content.Shared.Inventory.Events;
+
+namespace Content.Client.Clothing;
+
+/// <summary>
+///     Made by BL02DL from _LostParadise
+/// </summary>
+
+public sealed class NightVisionSystem : SharedNightVisionSystem
+{
+    [Dependency] private readonly IOverlayManager _overlayMan = default!;
+    [Dependency] private readonly ILightManager _lightManager = default!;
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+
+    private NightVisionOverlay _overlay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRestart);
+        SubscribeLocalEvent<NightVisionComponent, GotUnequippedEvent>(OnGotUnequipped);
+
+        _overlay = new(this);
+    }
+
+    public NightVisionComponent? GetNightComp()
+    {
+        var playerUid = EntityUid.Parse(_playerManager.LocalPlayer?.ControlledEntity.ToString());
+        var slot = _entityManager.GetComponent<InventorySlotsComponent>(playerUid);
+        _entityManager.TryGetComponent<NightVisionComponent>(slot.SlotData["eyes"].HeldEntity, out var nightvision);
```

Comment:
Why in the name of all that's unholy does this convert an entity uid to a string and parse it back?!

Below is a better example. It should work, but it was made in webedit and as such may contain a mistake/mistype/whatever.
```suggestion
        var playerUid = _playerManager.LocalPlayer?.ControlledEntity;
        if (!TryComp<InventorySlotsComponent>(playerUid, out var slot) || !TryComp<NightVisionComponent>(slot.SlotData["eyes"].HeldEntity, out var nightvision))
            return null;
```